cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -120,13 +120,13 @@ protected void startUp() throws Exception {
     LoggingContextAccessor.setLoggingContext(new ServiceLoggingContext(Id.Namespace.SYSTEM.getId(),
                                                                        Constants.Logging.COMPONENT_NAME,
                                                                        Constants.Service.APP_FABRIC_HTTP));
-    notificationService.start();
-    schedulerService.start();
-    applicationLifecycleService.start();
-    systemArtifactLoader.start();
-    programRuntimeService.start();
-    streamCoordinatorClient.start();
-    programLifecycleService.start();
+    notificationService.startAndWait();
+    schedulerService.startAndWait();
+    applicationLifecycleService.startAndWait();
+    systemArtifactLoader.startAndWait();
+    programRuntimeService.startAndWait();
+    streamCoordinatorClient.startAndWait();
+    programLifecycleService.startAndWait();
```

Comment:
have to wait for the to start in order to get the exception, but I can start them all at once and then wait for them all